File: multiplication_table.cpp
#include <iostream>
using namespace std;

void printTable(int num) {
    cout << "Multiplication Table for " << num << ":" << endl;

    for (int i = 1; i <= 12; i++) {
        cout << num << "  x  " << i << "  =  " << (num * i) << endl;
    }
}

void printTablesUpToN(int n) {
    for (int i = 1; i <= n; i++) {
        printTable(i);

        if (i < n) {
            cout << "---------------------------" << endl;
        }
    }
}

int main() {
    int number;

    cout << "Enter a number: ";
    cin >> number;

    printTable(number);

    cout << endl;

    int n;
    cout << "Enter N: ";
    cin >> n;

    if (n <= 0) {
        cout << "Error: N must be a positive integer." << endl;
        return 0;
    }

    printTablesUpToN(n);

    return 0;
}

File: Multiplication_table.js
// Multiplication Table Generator

const readlineSync = require('readline-sync');

function printSingleTable(num) {
  console.log("Multiplication Table for " + num + ":");

  for (let i = 1; i <= 12; i++) {
    console.log(num + "  x  " + i + "  =  " + (num * i));
  }
}

function printTablesUpToN(n) {
  for (let num = 1; num <= n; num++) {
    printSingleTable(num);
    console.log("---------------------------");
  }
}

function main() {
  // ---------- PART A: Single Table ----------
  const num = readlineSync.questionInt("Enter a number: ");

  console.log("");
  printSingleTable(num);

  // ---------- PART B: Tables from 1 to N ----------
  const n = readlineSync.questionInt("\nEnter N to print tables from 1 to N: ");

  if (n <= 0) {
    console.log("Error: Please enter a positive number.");
    return;
  }

  console.log("");
  printTablesUpToN(n);
}

main();